src/Message/Response/AbstractRabobankResponse.php

```diff
@@ -58,10 +58,9 @@ protected function validateSignature()
         if (!isset($this->data['signature'])) {
             return;
         }
-        
-        $signatureData = $this->data;
-        unset($signatureData['signature']);
-        unset($signatureData['timestamp']);
+
+        $allowedKeys = array_flip(['order_id', 'status']);
+        $signatureData = array_intersect_key($this->data, $allowedKeys);
```

Comment:
see also: https://github.com/thephpleague/omnipay-rabobank/pull/14
As I understand it, this function is used for request, response and status. The aparams for these are different though, that's why they used this particular method before. I quote a dev: "To make it more future-proof, we need some way to let the response object tell us what fields to use to check the signature."

so in short, are we sure this will work in all three situations?

Reply:
Well this will be a better option than what it currently is, if you want it for a more future-proof then this might be a way to go. Since Rabobank wil not change the signature as it currently is, then they must bring a new feature. (except if it is an security issue)

Reply:
Fixed per response can now have there own allowedKeys. Also for the payment_brands call that returns a call will get validated, so when there is a redirectUrl it will skip and do the old signatureValidation